src/recipes/editor-plugin/inspector-plugins.md

```diff
@@ -0,0 +1,342 @@
+}
+```
+
+After that, we need to add an implementation for the trait `IEditorProperty`:
+
+```rust
+#[godot_api]
+impl IEditorProperty for RandomIntEditor {
+    fn enter_tree(&mut self) {
+        // Create button element
+        let mut button = Button::new_alloc();
+        // Add handler for this button, handle_press will be define in another impl
+        button.connect(
+            "pressed".into(),
+            self.base().callable("handle_press".to_godot()),
+        );
+        button.set_text("Randomize".into());
+        // Save pointer to the button into struct
+        self.button = Some(button.clone());
+        self.base_mut().add_child(button.to_variant().to());
+    }
+
+    fn exit_tree(&mut self) {
+        // Remove element from inspector when this plugin unmount:
+        if let Some(button) = self.button.take() {
+            self.base_mut().remove_child(button.to_variant().to());
+        } else {
+            // Log error if button disappeared before
+            godot_error!("Button wasn't found in exit_tree");
+        }
+    }
+}
+```
+
+Let's add a handler for the button:
+
+```rust
+#[godot_api]
+impl RandomIntEditor {
+    #[func]
+    fn handle_press(&mut self) {
+        // Update value by button click
+        // - take prop name, randomize number
+        // - send prop name and random number to godot engine for update value
+        // - update button text
+        let property_name = self.base().get_edited_property();
+        let num = rand::thread_rng().gen_range(0..100);
+        godot_print!("Randomize! {} for {}", num, property_name);
+        self.base_mut()
+            .emit_changed(property_name, num.to_variant().to());
+        if let Some(mut button) = self.button.clone() {
+            let text = format!("Randomize: {}", num);
+            button.set_text(text.into());
+        } else {
+            // Print error of something went wrong
+            godot_error!("Button wasn't found in handle_press");
+        }
+    }
+}
+```
+
+
+## Add Inspector plugin
+
+Now we need to connect this editor to fields with an integer type.
+To do this, we need to create an EditorInspectorPlugin.
+
+```rust
+#[derive(GodotClass)]
+#[class(tool, init, base=EditorInspectorPlugin)]
+struct RandomInspectorPlugin {
+    base: Base<EditorInspectorPlugin>,
+}
+```
+
+IEditorInspectorPlugin implementation:
+
+```rust
+#[godot_api]
+impl IEditorInspectorPlugin for RandomInspectorPlugin {
+      fn parse_property(
+        &mut self,
+        _object: Gd<Object>, // object that inspecting
+        value_type: VariantType,
+        name: GString,
+        _hint_type: global::PropertyHint,
+        _hit_name: GString,
+        _flags: global::PropertyUsageFlags,
+        _wide: bool,
+    ) -> bool {
+        if value_type == VariantType::INT {
+            self.base_mut()
+                .add_property_editor(name, RandomIntEditor::new_alloc().to_variant().to());
+            return true;
+        }
+
+        false
+    }
```

Comment:
Also here, I'd use the official names (see earlier comment).

And the `to_variant().to()` can be `upcast()`.